File: web/lab1/server/Area.php
<?php

class Area {
    private $radius;
    private bool $isValid;
    private string $warning;

    public function __construct($radius) {
        $this->radius = $radius;
        $this->validateRadius();
    }

    public function checkHit(Point $point) {
        if ($point->isValid() && $this->isValid) {
            $result = 'hit';
            $x = (int)$point->getX();
            $y = (double)$point->getY();

            if (($x > 0 and $y > 0) or
                ($x > 0 and $y < 0 and 2 * $x + $y > $this->radius) or
                ($x < 0 and $y > 0 and sqrt($x * $x + $y * $y) > $this->radius / 2) or
                ($x < 0 and $y < 0 and $x > -$this->radius and $y > -$this->radius / 2)) $result = 'miss';

            return $result;
        }
        else return array_merge($point->getWarnings(), (array) $this->warning)[0];
    }

    private function validateRadius(): void
    {
        $this->isValid = false;
        if (!is_numeric($this->radius)){
            $this->warning = "радиус должен быть числом";
            return;
}
        if ((double) $this->radius - (int) $this->radius){
            $this->warning = "радиус должен быть целым числом";
            return;
        }
        if (!(1 <= $this->radius && $this->radius <= 5)){
            $this->warning = "неверный диапазон радиуса";
            return;
        }
        $this->radius = (int) $this->radius;
        $this->isValid = true;
    }

    public function getRadius()
    {
        return $this->radius;
    }

    public function isValid(): bool
    {
        return $this->isValid;
    }

    public function getWarning(): string
    {
        return $this->warning;
    }


}

File: web/lab2/src/main/java/com/radiad/lab2/Area.java
package com.radiad.lab2;

public class Area {
    private double radius;
    private String stringRadius;
    private boolean isValid;
    private String warning;

    public Area(String radius) {
        stringRadius = radius;
        validateRadius();
    }

    public String checkHit(Point point) {
        String result;
        if (point.isValid() && isValid) {
            result = "miss";
            double x = point.getX();
            double y = point.getY();

            if ((x >= 0 && y >= 0 && x + y <= radius / 2) ||
                    (x <= 0 && y <= 0 && Math.sqrt(x * x + y * y) <= radius / 2) ||
                    (x >= 0 && y <= 0 && x <= radius / 2 && y >= -radius)) result = "hit";

            return result;
        }
        else if (!point.isValid()) return point.getWarning();
        else return warning;
    }

    private void validateRadius() {
        isValid = false;
        try {
            radius = Double.parseDouble(stringRadius);
            if (!(1 <= radius && radius <= 5)){
                warning = "неверный диапазон радиуса";
                return;
            }
        } catch (NumberFormatException e) {
            warning = "радиус должен быть целым числом";
            return;
        }
        isValid = true;
    }

    public double getRadius() {
        return radius;
    }

    public boolean isValid() {
        return isValid;
    }

    public String getWarning() {
        return warning;
    }
}
